### ISSUE TITLE
[feat] Create `<rh-dropdown-menu>` element

### ISSUE THREAD

**Issue by wesruv** [RedHat-UX/red-hat-design-system#425]
### Description

Make a dropdown that can accept links or buttons that has the correct a11y properties.

PFE-Dropdown was a solid attempt at this but the aria attributes are setup incorrectly, see: https://github.com/patternfly/patternfly-elements/issues/2054#issuecomment-1178799554

I don't know anything about Patternfly React's dropdown, but it looks like it's following the same design and has features I'd be interested in: https://www.patternfly.org/v4/components/dropdown

One pattern that PFE didn't cover that it looks like PF React Dropdown did was the dropdown should close if it loses focus, and you should only be able to have one open at a time.

### Acceptance Criteria

- [ ] Design done
- [ ] Development done

### Image

![image](https://user-images.githubusercontent.com/5607236/178032549-66cc417e-1ebd-4d8b-abfa-1058c1b7d49a.png)

### Link to design doc

https://www.patternfly.org/v4/components/dropdown/design-guidelines

**Comment by wesruv** [comment]
I switched up the labels to what I think sounded right.

**Comment by bennypowers** [comment]
See also https://github.com/patternfly/patternfly-elements/issues/2049

**Comment by markcaron** [comment]
Blocked by PFE 3.0 release (https://github.com/patternfly/patternfly-elements/pull/2596)

**Comment by hellogreg** [comment]
Exists as `rh-menu`

**Comment by markcaron** [comment]
@adamjohnson @bennypowers @zeroedin, we should look into what's left to wrap up `<rh-menu>` to close this loop.

**Comment by marionnegp** [comment]
@markcaron, since the design here is referencing PFv4, do we want to revisit it quickly to make sure it's still the design direction we want to use?

**Comment by adamjohnson** [comment]
Indeed, I was just going to mention that we're missing a Figma for this component.

Also relevant for the design phase: `<rh-dropdown>` is likely made for freeform content (eg: primary nav dropdowns) whereas `<rh-menu>` follows the APG [Menu Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/):

> A [menu](https://w3c.github.io/aria/#menu) is a widget that offers a list of choices to the user, such as a set of actions or functions. Menu widgets behave like native operating system menus, such as the menus that pull down from the menubars commonly found at the top of many desktop application windows.

Looking at `<rh-menu>` right now, it looks like this:

![Screenshot of rh-menu and four menu items](https://github.com/user-attachments/assets/9754a0db-092f-4270-a313-72a9bbf42213)

`<rh-menu>` is not fully baked.

**Comment by coreyvickery** [comment]
@marionnegp Design looks good to me.
